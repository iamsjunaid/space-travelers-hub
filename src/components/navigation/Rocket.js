import React from 'react';
import { useDispatch } from 'react-redux';
import PropTypes from 'prop-types';
import { reserveRocket, cancelReserve } from '../../redux/rockets/rocketsSlice';
import '../../styles/Rocket.css';

function Rocket({
  id, name, description, image, reserved,
}) {
  const dispatch = useDispatch();
  return (
    <li className="rocket">
      <img src={image} alt={name} />
      <div>
        <h3>{name}</h3>
        <div>
          <p>
            {reserved && <span className="reserved">Reserved</span>}
            {' '}
            <span>
              {description}
            </span>
          </p>
        </div>
        {!reserved && (
          <button className="reserve-rocket" type="button" onClick={() => dispatch(reserveRocket(id))}>
            Reserve Rocket
          </button>
        )}
        {reserved && (
          <button className="cancel-rocket" type="button" onClick={() => dispatch(cancelReserve(id))}>
            Cancel Reservation
          </button>
        )}
      </div>
    </li>
  );
}

Rocket.propTypes = {
  id: PropTypes.number.isRequired,
  name: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
  image: PropTypes.string.isRequired,
  reserved: PropTypes.bool.isRequired,
};

export default Rocket;
