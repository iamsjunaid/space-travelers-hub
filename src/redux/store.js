import { configureStore } from '@reduxjs/toolkit';
import rocketsReducer from './rockets/rocketsSlice';
import dragonsReducer from './dragons/dragonsSlice';

const store = configureStore({
  reducer: {
    rockets: rocketsReducer,
    dragons: dragonsReducer,
  },
});

export default store;
