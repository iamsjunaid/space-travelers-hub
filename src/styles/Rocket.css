.rocket {
  display: flex;
  gap: 0.5em;
}

.rocket img {
  width: 350px;
  object-fit: contain;
}

.rocket button {
  border-radius: 5px;
  padding: 10px;
  cursor: pointer;
  width: 12em;
}

.reserve-rocket {
  background-color: #007bff;
  border: none;
  color: #fff;
}

.cancel-rocket {
  background-color: #fff;
  color: #000;
  border: 1px solid #000;
}

.rocket h3 {
  font-weight: normal;
}

.reserved {
  background-color: #007bff;
  color: white;
  text-align: center;
  border-radius: 5px;
  padding: 5px;
  font-size: 12px;
}
