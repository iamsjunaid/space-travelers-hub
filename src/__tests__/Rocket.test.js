import '@testing-library/jest-dom';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import React from 'react';
import store from '../redux/store';
import Rocket from '../components/navigation/Rocket';

describe('User Interactions test', () => {
  const rocket = [
    {
      id: 1,
      rocket_name: 'Falcon 9',
      description: 'A two-stage rocket designed and manufactured by SpaceX.',
      image: 'falcon9.jpg',
    },
  ];

  it('will test user interaction', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <Rocket rocket={rocket} />
        </Provider>,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
