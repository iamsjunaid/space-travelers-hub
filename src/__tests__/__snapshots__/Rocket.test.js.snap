// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User Interactions test will test user interaction 1`] = `
<li
  className="rocket"
>
  <img />
  <div>
    <h3 />
    <div>
      <p>
         
        <span />
      </p>
    </div>
    <button
      className="reserve-rocket"
      onClick={[Function]}
      type="button"
    >
      Reserve Rocket
    </button>
  </div>
</li>
`;
